### PR TITLE
Remove the fanInOnWithfallBack method and the related system property.

### PR DESCRIPTION
**Pull request by varshavaradarajan**

### CODE REVIEW

File: base/src/com/thoughtworks/go/util/SystemEnvironment.java [hunk 1]
```diff
@@ -678,13 +677,6 @@ public int getTfsSocketTimeout() {
         return Integer.parseInt(getPropertyImpl(TFS_SOCKET_TIMEOUT_PROPERTY, String.valueOf(TFS_SOCKET_TIMEOUT_IN_MILLISECONDS)));
     }
 
-    public boolean enforceFanInFallbackBehaviour() {
-        if (enforceFanInFallbackTriangle == null) {
```
Comment (jyotisingh) [inline base/src/com/thoughtworks/go/util/SystemEnvironment.java]: Delete unused variable `enforceFanInFallbackTriangle`

File: server/src/com/thoughtworks/go/server/service/AutoBuild.java [hunk 2]
```diff
@@ -108,21 +106,6 @@ public boolean shouldCheckWhetherOlderRunsHaveRunWithLatestMaterials() {
     public void notifyPipelineNotScheduled(PipelineConfig pipelineConfig) {
     }
 
-    private MaterialRevisions fanInOnWithFallback(MaterialRevisions originalMaterialRevisions, CruiseConfig cruiseConfig, CaseInsensitiveString targetPipelineName) {
```
Comment (jyotisingh) [inline server/src/com/thoughtworks/go/server/service/AutoBuild.java]: remove unused variable `serverHealthService`

**Comment by varshavaradarajan** [comment]
@jyotisingh , do we have to wait for the functional tests? If you think this is fine, can you please merge it?

**Comment by jyotisingh** [comment]
What's the hurry? :)  let's wait for functional tests unless you have checked that the flag is not used in any of those tests. Will merge in the morning.

**Comment by varshavaradarajan** [comment]
https://build.gocd.org/go/pipelines/value_stream_map/trigger/1635

build-windows/build-non-server passed too - https://build.gocd.org/go/pipelines/build-windows-PR/1863/build-non-server/1 

@arvindsv , @jyotisingh wanted your okay on this.

**Pull request merged**